### DISCUSSION TITLE
How to compile v programs for mcu targets ，like cortex-m3

### DISCUSSION THREAD

**Discussion by kekemuyu** [vlang/v#11621] [Ideas]
I want to write programs with v on embeded system, but I dont know how to compile it.

**Comment by alula** [comment]
Freestanding support for V is still I early state but you can take look at vOS: https://github.com/vlang/vos

**Comment by dumblob** [comment]
Related: [`volatile` missing in *V*](https://github.com/vlang/v/issues/2761#issuecomment-621329529 )

**Comment by dumblob** [comment]
`volatile` became part of *V* - see https://github.com/vlang/v/pull/11940 .

**Reply by fxyc87** [reply]
Not at the moment

**Reply by dumblob** [reply]
Now it really is part of `V` :wink:.
